### PR TITLE
Add --reruns flag

### PR DESCRIPTION
**Pull request by RagnarGrootKoerkamp**

### CODE REVIEW

File: experiments_main.cpp [hunk 1]
```diff
@@ -201,8 +214,62 @@ class ExperimentRunner {
...
-                    algorithms, dists);
+        apply_tuple(
+                [&](auto &algorithm, auto &dist) {
+                    std::cout << "Running " << algorithm.name << std::endl;
+
+                    // Run the algorithms FLAGS_reruns times, storing the distances and Spearman
+                    // coefficient computed in each run. The average and standard deviation of the
+                    // Spearman coefficients is reported, as well as the Spearman coefficient
+                    // obtained from using the median and average of the distances of all runs.
+                    vector<double> spearman_coefficients(FLAGS_reruns);
```
Comment (danieldanciu) [inline experiments_main.cpp]: should be `std::vector<>`. How come `vector` works?

File: experiments_main.cpp [hunk 2]
```diff
@@ -201,8 +214,62 @@ class ExperimentRunner {
...
+
+                    if (FLAGS_reruns > 1) {
+                        // Transpose of dists_per_run.
+                        Vec2D<double> runs_per_dist
+                                = new2D<double>(ingroup_pairs.size(), FLAGS_reruns);
+                        for (size_t i = 0; i < FLAGS_reruns; ++i)
+                            for (size_t j = 0; j < dists_per_run[i].size(); ++j)
+                                runs_per_dist[j][i] = dists_per_run[i][j];
+
+                        for (auto &distances : runs_per_dist)
```
Comment (danieldanciu) [inline experiments_main.cpp]: auto->double

For readability, the convention I used so far is to only use `auto` when the type is absolutely clear (e.g. with `make_unique`) or when the type is so complex that it reduces readability to write it (typically with iterators).

In this particular case, it saves teh code reader a lookup to figure out what type the distance actually is.

File: experiments_main.cpp [hunk 3]
```diff
@@ -201,8 +214,62 @@ class ExperimentRunner {
...
+                        std::cout << "\t"
+                                  << "Average  Corr.: " << avg << " \t (σ=" << sd
+                                  << ", n=" << FLAGS_reruns << ")" << std::endl;
+
+                        dist.resize(ingroup_pairs.size());
+
+
+                        for (size_t i = 0; i < ingroup_pairs.size(); ++i)
+                            dist[i] = median(runs_per_dist[i]);
+                        auto sc_on_med_dist = spearman(edit_dists, dist);
```
Comment (danieldanciu) [inline experiments_main.cpp]: double

File: experiments_main.cpp [hunk 4]
```diff
@@ -201,8 +214,62 @@ class ExperimentRunner {
...
+                        std::cout << "\t"
+                                  << "SC on med dist: " << sc_on_med_dist << std::endl;
+
+                        for (size_t i = 0; i < ingroup_pairs.size(); ++i)
+                            dist[i] = std::accumulate(begin(runs_per_dist[i]),
+                                                      end(runs_per_dist[i]), 0.0)
+                                    / FLAGS_reruns;
+                        auto sc_on_avg_dist = spearman(edit_dists, dist);
```
Comment (danieldanciu) [inline experiments_main.cpp]: double

**Review by danieldanciu — APPROVED** [review]

File: experiments_main.cpp (hunk 1 above)
Comment (RagnarGrootKoerkamp) [reply experiments_main.cpp]: It was you :P fixed now
https://github.com/ratschlab/Project2020-seq-tensor-sketching/blame/master/util/spearman.hpp#L7

File: experiments_main.cpp (hunk 2 above)
Comment (RagnarGrootKoerkamp) [reply experiments_main.cpp]: I like `auto` personally, but I'll try not to  :)

**Pull request merged**